### ISSUE TITLE
Apply External Perimeter Extrusion Width to internal perimeters that are visible from above/below too

### ISSUE THREAD

**Issue by printrboy** [alexrj/Slic3r#2433]
Love the new option in 1.2 which allows to print external perimeters at narrower diameter while the inner perimeters not.
This, combined with "top solid infill" width allows me to print some pretty darn good looking models without waiting for days for print to finish.

But I think there's a bit of problem of how it does it now. Outermost perimeters are not the only ones which are "exposed" or visible, or "external", yet only they are treated as such.
A perimeter can be "behind" another perimeter but still be visible from top or bottom.

Here's an example.
Let's say your printing something like a cube.
At top you will have say 5 perimeters exposed and an infill.

<a href="http://imgur.com/uBVvBCZ"><img src="http://i.imgur.com/uBVvBCZ.png" title="source: imgur.com" /></a>

In this image, you see the side of the topmost layer of the cube, green is the top solid infill, red is the outermost perimeter and blues are the rest perimeters.

I can currently set the greens and blues to be narrower unlike the inner perimeters and inner (not outermost) infills. But I can't affect the blue ones. Only if I set every single perimeter to be narrow which adds a lot to the print time.

And flat tops is not the only case where this is a problem, so making all the infills of the topmost layer use the value of the "external perimeter" won't be a solution.

Because here's another example.
Lets say you're printing something like this (side cutout):
<a href="http://imgur.com/DkOlAip"><img src="http://i.imgur.com/DkOlAip.png" title="source: imgur.com" /></a>

Zooming it, and hiding everything except perimeters:
<a href="http://imgur.com/HnN5xdE"><img src="http://i.imgur.com/HnN5xdE.png" title="source: imgur.com" /></a>

See where I'm going with this?
This
<a href="http://imgur.com/r9eNj9M"><img src="http://i.imgur.com/r9eNj9M.png" title="source: imgur.com" /></a>

Should be like this instead

<a href="http://imgur.com/S6kHLxV"><img src="http://i.imgur.com/S6kHLxV.png" title="source: imgur.com" /></a>

Otherwise it beats the purpose of the "external perimeters" value for many 3d models.

And again, making all perimeters narrow isn't a good solution, because imagine having 2 more internal perimeters (to the left) of the existing two in the above image. It will increase print time, which is the reason "external perimeters" option was added to combat.

In short, not only outermost perimeters should be treated as "external perimeters".

**Comment by lordofhyphens** [comment]
@alexrj says he's not changing it, I'd probably not accept a PR to change it either without a heck of a lot of convincing (or I'll defer to @alexrj :) )
Closing.

**Comment by bubnikv** [comment]
I believe I understand both points of view. I see some reasons to implement the feature proposed.

Ssome nozzles have quite a sharp tip, some have a larger flat tip. The flat nozzles iron out the thick extrudates nicely, not so the nozzles with a sharp tip. I suspect Alex has printers with quite a flat nozzle tip due to the way Slic3r sets the default extrusion widths when set to zero in the UI.

Then there are the gaps created at a top surface with wider extrudates. Depending on the material, print speed, nozzle temperature and the nozzle shape, these gaps may or may not be filled by the excess material pressed and squashed by the nozzle tip. I suppose with some printers or some materials it is simply impossible to fill in these gaps, so the proposal to use a narrow extrudate on top surfaces makes sense.

Vojtech

**Comment by lordofhyphens** [comment]
@bubnikv I'm of the opinion that if your nozzle isn't flat enough to flatten the surface appropriately then you're running extrusion widths that are too wide :) 

But for a UI burden of a checkbox (because making this the default is going to increase print times even more and I'm not convinced it's necessary or appropriate) makes it feasible to accept a PR for it.

**Comment by bubnikv** [comment]
The wide extrusion may be ok for surfaces, which are hidden. Then the uneven surface left by the too narrow nozzle are hidden by the layer above. As long as the layers bond well, all is bright and the thick extrusion serves its purpose to give the object its rigidity.
